src/main/ui/app/route/Show.jsx

```diff
@@ -74,7 +76,7 @@ export default class Show extends Component {
             <Label bsStyle={statusClass}>{status}</Label>
             {addon.tags ?
              addon.tags.map(t =>
-                 <Label bsStyle="default">{t}</Label>
+                 <Link to={`/search?&tag=${t}`}><Label bsStyle="default">{t}</Label></Link>
```

Comment:
I'll add to our list of tasks. This is a minor one and hopefully can be focused on and discussed upon later :)